client/actions/events/api.ts

```diff
@@ -113,99 +112,49 @@ const unspike = (events) => (
 
-            if (ids.length <= chunkSize) {
-                itemIds = ids;
+            if (params.ids.length <= chunkSize) {
+                itemIds = params.ids;
             } else {
                 // chunk the requests
                 const requests = [];
 
-                for (let i = 0; i < Math.ceil(ids.length / chunkSize); i++) {
+                for (let i = 0; i < Math.ceil(params.ids.length / chunkSize); i++) {
                     const args = {
                         // eslint-disable-next-line no-undef
-                        ...arguments[0],
-                        ids: ids.slice(i * chunkSize, (i + 1) * chunkSize),
+                        ...params,
+                        ids: params.ids.slice(i * chunkSize, (i + 1) * chunkSize),
                     };
 
                     requests.push(dispatch(self.query(args)));
                 }
-                // flattern responses and return a response-like object
+                // flatten responses and return a response-like object
                 return Promise.all(requests).then((responses) => (
                     Array.prototype.concat.apply([], responses)
                 ));
             }
         }
 
-        return planningApis.events.search({
-            item_ids: itemIds,
-            name: advancedSearch?.name,
-            tz_offset: getTimeZoneOffset(),
-            full_text: fulltext,
-            anpa_category: advancedSearch.anpa_category,
-            subject: advancedSearch.subject,
-            posted: advancedSearch.posted,
-            place: advancedSearch.place,
-            language: advancedSearch.language,
-            state: advancedSearch.state,
-            spike_state: spikeState,
-            include_killed: includeKilled,
-            date_filter: advancedSearch?.dates?.range,
-            start_date: advancedSearch?.dates?.start,
-            end_date: advancedSearch?.dates?.end,
-            start_of_week: appConfig.start_of_week,
-            slugline: advancedSearch?.slugline,
-            recurrence_id: recurrenceId,
-            reference: advancedSearch?.reference,
-            source: advancedSearch?.source,
-            location: typeof advancedSearch.location === 'string' ?
-                advancedSearch.location :
-                advancedSearch.location?.name,
-            calendars: calendars,
-            no_calendar_assigned: noCalendarAssigned,
-            only_future: onlyFuture,
-            page: page,
-            max_results: maxResults,
-            filter_id: filter_id || selectors.main.currentSearchFilterId(getState()),
-        })
+        return planningApis.events.search(eventParamsToSearchParams({
+            ...params,
+            itemIds: itemIds,
+            timezoneOffset: getTimeZoneOffset(),
+            spikeState: params.spikeState ?? 'draft',
+            onlyFuture: params.onlyFuture ?? true,
+            page: params.page ?? 1,
+            maxResults: params.maxResults ?? MAIN.PAGE_SIZE,
```

Comment:
just wondering why not handle all those defaults in the helper

Reply:
These were left over from the previous code while migrating to the helpers. I'll create a new commit to move these